doc/development/architecture.rst

```diff
@@ -0,0 +1,315 @@
+
+The following diagram illustrates the workflow for two common solver types for a conjugate heat transfer scenario:
+
+.. mermaid::
+
+    %%{init: {'flowchart': { 'htmlLabels': true, 'wrap': true }}}%%
+    flowchart TB
+        subgraph setup1 ["setup "]
+            direction TB
+            S1A["Initialize Fields"]
+        end
+
+        subgraph setup2 ["setup"]
+            direction TB
+            S2A["Initialize Temperature Field"]
+        end
+
+        subgraph S1 ["non-thermal-fluid solver"]
+            direction TB
+            S1C["Momentum Predictor"]
+            S1C --> S1E["Solve Energy Equation"]
+            S1E --> S1F["Pressure Corrector<br/>PISO Loop"]
+            S1F --> S1J["Update turbulence Model"]
+        end
+
+        subgraph S2 ["thermal-solid solver"]
+            direction TB
+            S2B["Solve Energy Equation"] --> S2D["Update Solid Properties"]
+        end
+
+        setup1 --> S1C
+        setup2 --> S2B
+        S1E --> S2B
+        S2B --> S1E
+
+
+The diagram shows two selected solvers: a non-thermal-fluid solver and a thermal-solid solver.
+Each solver has its own setup phase and computational steps, with interactions between the two solvers for energy exchange.
+
+The solution steps of each solver provide the dependencies and the runtime system constructs a DAG to determine the correct execution order based on these dependencies.
+
+.. code-block:: python
+
+    # 'model' is a placeholder for the model dependency object required by each step.
+    model = ...  # placeholder for model dependency
+    @model("fluidSolver")
+    class DomainA:
+        @step(order=1, model)
+        def momentum(self, ctrl): pass
+        @step(order=2, model)
+        def solve_energy(self, ctrl): pass
+        @step(order=3, model)
+        def pressure_corrector(self, ctrl): pass
+        @step(order=4, model)
+        def update_turbulence(self, ctrl): pass
+
+    @model("solidSolver")
+    class DomainB:
+        @step(order=1, model)
+        def solve_energy(self, ctrl): pass
+        @step(order=2, model)
+        def update_solid_properties(self, ctrl): pass
+
+A solver is defined as a class with multiple steps, each step has to declare its dependencies.
+Additional models can be executed before and after each step to modify the behavior and add additional physics.
+
+This allows to easily extend existing solvers with new physics without modifying the core solver implementation and promotes code reuse.
+
+.. note::
+
+    The implementation is still work in progress but only details may change.
+
+Modular Solver Architecture
+---------------------------
+
+The modular solver architecture briefly touched on in the previous section is a key feature of FoamAdapter and allows for flexible composition of complex multi-physics simulations.
+An example of the modular solver architecture is illustrated in the following diagram:
```

Comment:
My understanding is

```suggestion
The modular solver architectures allows to extend individiual solver by adding addititional physics modules.
This is illustrated in the following diagram, where the fluid solver is extended by three physics submodules.
```

Then proceed to discuss how and why individual modules interact with the other components.